src/Analyzers/CSharp/Tests/RemoveRedundantElseStatement/RemoveRedundantElseStatementTests.cs

```diff
@@ -0,0 +1,843 @@
+
+        [Fact]
+        public async Task TestRedundantElseFix_NestedIfStatementsInSwitch()
+        {
+            await VerifyCS.VerifyCodeFixAsync("""
+                using System;
+
+                class C
+                {
+                    int M(int n) 
+                    {
+                        if (true)
+                        {
+                            switch (n)
+                            {
+                                default:
+                                    if (true)
+                                    {
+                                        break;
+                                    }
+                                    [|else|]
+                                    {
+                                        break;
+                                    }
+                            }
+
+                            return 0;
+                        }
+                        [|else|]
+                        {
+                            return 0;
+                        }
+                    }
+                }
+                """, """
+                using System;
+                
+                class C
+                {
+                    int M(int n) 
+                    {
+                        if (true)
+                        {
+                            switch (n)
+                            {
+                                default:
+                                    if (true)
+                                    {
+                                        break;
+                                    }
+
+                                    break;
+                            }
+
+                            return 0;
+                        }
+
+                        return 0;
+                    }
+                }
+                """);
+        }
+
+        [Fact]
+        public async Task TestRedundantElseFix_GlobalStatement()
+        {
+            var test = new VerifyCS.Test
+            {
+                TestCode = """
+                    using System;
+                    
+                    if (false)
+                    {
+                        return;
+                    }
+                    [|else|]
+                    {
+                        Console.WriteLine("Success");
+                    }
+                    """,
+                FixedCode = """
+                    using System;
+                    
+                    if (false)
+                    {
+                        return;
+                    }
+                    
+                    Console.WriteLine("Success");
+
+                    """,
+                LanguageVersion = LanguageVersion.CSharp9
+            };
+
+            test.ExpectedDiagnostics.Add(
+                // /0/Test0.cs(3,1): error CS8805: Program using top-level statements must be an executable.
+                DiagnosticResult.CompilerError("CS8805").WithSpan(3, 1, 10, 2)
+            );
```

Comment:
You can avoid this diagnostic be setting OutputKind. See the following test for example on how to do that:

https://github.com/dotnet/roslyn/blob/c31875bdfde1e64b1879212e4b8f974973704b08/src/Analyzers/CSharp/Tests/UseDeconstruction/UseDeconstructionTests.cs#L859-L878

Reply:
Thanks 😄